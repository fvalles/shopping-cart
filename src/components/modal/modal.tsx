import React, { FunctionComponent, ReactElement } from 'react';
import ModalWrapper from 'react-modal';

/**
 * Types
 */

interface ModalProps {
  /** Modal content components */
  children: ReactElement;
  /** Boolean describing if the modal should be shown or not. */
  isOpen: boolean;
  /** String indicating how the content container should be announced to screenreaders */
  label: string;
  /** String testId that renders a data-testid attribute in the DOM, used for testing. */
  testId: string;
}

/**
 * Constants
 */

const MODAL_STYLES = {
  content: {
    borderRadius: 4,
    display: 'flex',
    height: 'calc(100% - 64px)',
    left: '50%',
    maxHeight: 648,
    maxWidth: 1088,
    padding: 0,
    top: '50%',
    transform: 'translate(-50%, -50%)',
    width: 'calc(100% - 64px)',
  },
};

/**
 * Modal
 */

export const Modal: FunctionComponent<ModalProps> = ({
  children,
  isOpen,
  label,
  testId,
}) => (
  <ModalWrapper
    ariaHideApp={false}
    contentLabel={label}
    isOpen={isOpen}
    shouldCloseOnOverlayClick={false}
    shouldCloseOnEsc={false}
    style={MODAL_STYLES}
    testId={testId}>
    {children}
  </ModalWrapper>
);
