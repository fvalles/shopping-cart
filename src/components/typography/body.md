Basic body text component with default color="body":

```jsx
<Body>Lorem ipsum</Body>
```

Body text component with color="primary":

```jsx
<Body color="primary">Lorem ipsum</Body>
```
