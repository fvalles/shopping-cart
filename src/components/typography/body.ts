import styled from 'styled-components';
import { TypographyProps } from './common';

export const Body = styled.p<TypographyProps>`
  color: ${({ color = 'body', theme }) => theme.Colors[color]};
  font-family: ${({ theme }) => theme.Fonts.regular}, 'Avenir Next', 'Segoe UI',
    sans-serif;
  font-size: 13px;
  font-weight: 400;
  letter-spacing: -0.01em;
  line-height: 18px;
`;
