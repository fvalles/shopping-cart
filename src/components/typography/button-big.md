Basic button big text component with default color="primary":

```jsx
<ButtonBig>Lorem ipsum</ButtonBig>
```

Button big text component with color="summaryText":

```jsx
<ButtonBig color="summaryText">Lorem ipsum</ButtonBig>
```
