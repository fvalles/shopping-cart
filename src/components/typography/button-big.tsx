import styled from 'styled-components';
import { TypographyProps } from './common';

export const ButtonBig = styled.span<TypographyProps>`
  color: ${({ color = 'primary', theme }) => theme.Colors[color]};
  font-family: ${({ theme }) => theme.Fonts.regular}, 'Avenir Next', 'Segoe UI',
    sans-serif;
  font-size: 20px;
  font-weight: bold;
  line-height: 25px;
`;
