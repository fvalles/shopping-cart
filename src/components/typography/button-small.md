Basic button small text component with default color="white":

```jsx
<ButtonSmall>Lorem ipsum</ButtonSmall>
```

Button small text component with color="summaryText":

```jsx
<ButtonSmall color="summaryText">Lorem ipsum</ButtonSmall>
```
