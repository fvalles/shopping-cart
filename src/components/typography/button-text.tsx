import styled from 'styled-components';
import { TypographyProps } from './common';

export const ButtonText = styled.span<TypographyProps>`
  color: ${({ color = 'white', theme }) => theme.Colors[color]};
  font-family: ${({ theme }) => theme.Fonts.regular}, 'Avenir Next', 'Segoe UI',
    sans-serif;
  font-size: 16px;
  font-weight: bold;
  line-height: 14px;
`;
