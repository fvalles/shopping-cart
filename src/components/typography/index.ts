export { ButtonBig } from './button-big';
export { ButtonSmall } from './button-small';
export { H1 } from './h1';
export { H2 } from './h2';
export { H3 } from './h3';
export { ListItemBig } from './list-item-big';
export { ListItemMedium } from './list-item-medium';
export { ListItemSmall } from './list-item-small';
export { Span } from './span';
export { Subtitle } from './subtitle';
export { Title } from './title';
