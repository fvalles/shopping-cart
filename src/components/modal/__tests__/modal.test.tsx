import React from 'react';
import { ThemeProvider } from '@core/theme/theme-provider';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import { H1 } from '@components/typography';
import { Modal } from '../modal';

jest.mock('react-dom', () => ({
  ...jest.requireActual('react-dom'),
  createPortal: (node: any) => node,
}));

describe('modal component', () => {
  it('should not render if isOpen property is false', () => {
    const result = renderer
      .create(
        <ThemeProvider>
          <Modal isOpen={false} label="some label">
            <H1>Lorem ipsum</H1>
          </Modal>
        </ThemeProvider>,
      )
      .toJSON();

    expect(result).toBe(null);
    expect(result).toMatchSnapshot();
  });

  it('should render if isOpen property is true', () => {
    const result = renderer
      .create(
        <ThemeProvider>
          <section>
            <Modal isOpen label="some label">
              <H1>Lorem ipsum</H1>
            </Modal>
          </section>
        </ThemeProvider>,
      )
      .toJSON();

    const container = result.children[0];
    const modal = container.children[0];
    const modalLabel = modal.props['aria-label'];
    expect(modalLabel).toBe('some label');

    const modalTextContent = modal.children[0];
    expect(modalTextContent.type).toBe('h1');
    expect(modalTextContent.children[0]).toBe('Lorem ipsum');
    expect(result).toMatchSnapshot();
  });
});
