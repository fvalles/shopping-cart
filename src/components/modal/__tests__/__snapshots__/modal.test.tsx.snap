// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`modal component should not render if isOpen property is false 1`] = `null`;

exports[`modal component should render if isOpen property is true 1`] = `
.c0 {
  color: #212240;
  font-family: Avenir Medium,'Avenir Next','Segoe UI', sans-serif;
  font-size: 18px;
  -webkit-letter-spacing: -0.18px;
  -moz-letter-spacing: -0.18px;
  -ms-letter-spacing: -0.18px;
  letter-spacing: -0.18px;
  line-height: 24px;
}

<section>
  <div
    className="ReactModal__Overlay"
    onClick={[Function]}
    onMouseDown={[Function]}
    style={
      Object {
        "backgroundColor": "rgba(255, 255, 255, 0.75)",
        "bottom": 0,
        "left": 0,
        "position": "fixed",
        "right": 0,
        "top": 0,
      }
    }
  >
    <div
      aria-label="some label"
      aria-modal={true}
      className="ReactModal__Content"
      data-testid="some-test-id"
      onClick={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      onMouseUp={[Function]}
      role="dialog"
      style={
        Object {
          "WebkitOverflowScrolling": "touch",
          "background": "#fff",
          "border": "1px solid #ccc",
          "borderRadius": 4,
          "bottom": "40px",
          "display": "flex",
          "height": "calc(100% - 64px)",
          "left": "50%",
          "maxHeight": 648,
          "maxWidth": 1088,
          "outline": "none",
          "overflow": "auto",
          "padding": 0,
          "position": "absolute",
          "right": "40px",
          "top": "50%",
          "transform": "translate(-50%, -50%)",
          "width": "calc(100% - 64px)",
        }
      }
      tabIndex="-1"
    >
      <h1
        className="c0"
      >
        Lorem ipsum
      </h1>
    </div>
  </div>
</section>
`;
